syntax = "proto3";

option java_outer_classname = "ClassifierProto";
option objc_class_prefix = "KC";

service Classifier {
  rpc GetFeatureNames(DataID) returns (FeatureNames);
  rpc GetPrediction(Request) returns (ClassifierResult);
}

message DataID {
	int32 idx = 1;
}

message FeatureNames {
	string feature_names = 1;
}

message Request {
	DataID idx = 1;
	string request = 2;
}

message ClassifierResult{
	string label = 1;
}
